src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -0,0 +1,382 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * Adapter building static PHP files that will be cached by OPCache.
+ * This Adapter is in read-only if you use AdapterInterface methods.
+ * You can use the method "store" to build the cache file.
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class OpCacheAdapter implements AdapterInterface, LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    private $file;
+    private $values;
+    private $createCacheItem;
+
+    /** @var AdapterInterface */
+    private $fallbackPool;
+
+    /**
+     * @return CacheItemPoolInterface
+     */
+    public static function createOnPhp7($file, CacheItemPoolInterface $fallbackPool)
```

Comment:
But it does nothing on php 5... Do you have any suggestions? It's hard to name things ;)